### PR TITLE
Wildfly s2i migrated to use openshift legacy fp

### PR DESCRIPTION
**Pull request by jfdenise**
This is a refactoring of the WildFly s2i image to use wildfly-openshift-legacy-galleon-pack.

### CODE REVIEW

File: wildfly-modules/jboss/container/wildfly/galleon-wildfly/artifacts/opt/jboss/container/wildfly/galleon/wildfly-s2i-galleon-pack/pom.xml [hunk 1]
```diff
@@ -33,9 +33,25 @@
             <distribution>repo</distribution>
         </license>
     </licenses>
-    
+    <repositories>
+        <repository>
+            <id>jboss-public-snapshots</id>
+            <name>JBoss Public Snapshots Repository Group</name>
+            <url>https://repository.jboss.org/nexus/content/repositories/snapshots/</url>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-modules/jboss/container/wildfly/galleon-wildfly/artifacts/opt/jboss/container/wildfly/galleon/wildfly-s2i-galleon-pack/pom.xml:40]: You found it ;-), all is very preliminary till we have failing tests.

File: wildfly-modules/jboss/container/wildfly/galleon-wildfly/artifacts/opt/jboss/container/wildfly/galleon/wildfly-s2i-galleon-pack/wildfly-user-feature-pack-build.xml [hunk 2]
```diff
@@ -38,5 +38,10 @@
             <packages inherit="false"/>
             <default-configs inherit="false"/>
         </dependency>
+        <dependency group-id="org.jfdenise.demo" artifact-id="wildfly-openshift-legacy-galleon-pack">
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-modules/jboss/container/wildfly/galleon-wildfly/artifacts/opt/jboss/container/wildfly/galleon/wildfly-s2i-galleon-pack/wildfly-user-feature-pack-build.xml:41]: Sure.

File: wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" ?>
 
+<!-- The default config expects the wildfly-datasources-galleon-pack to be present in the provisioning config -->
```
Comment (jmesnil) [inline wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml:3]: Does that mean that the wildfly-cloud-legacy-galleon-pack as a (non-optional) dependency on wildfly-datasources-galleon-pack.

Does that mean that https://github.com/wildfly/wildfly-s2i/pull/173/files#diff-f98913fa22922d628a0ea4d0523af9a716ce1b51c888b7a81d8ce3ea30b3b9e0R75-R80 is always required to ensure both feature packs are provisioned?

File: wildfly-cloud-legacy-galleon-pack/wildfly-feature-pack-build.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<build xmlns="urn:wildfly:feature-pack-build:3.1" producer="org.wildfly:wildfly-cloud-legacy-galleon-pack">
```
Comment (jmesnil) [inline wildfly-cloud-legacy-galleon-pack/wildfly-feature-pack-build.xml:23]: related to my comment above: is wildfly-datasource-galleon-pack a dependency here?

File: wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml (hunk 3 above)
Comment (jmesnil) [inline wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml:3]: does that mean that wildfly-datasources-galleon-pack is a dependency of wildfly-cloud-legacy-galleon-pack?

File: wildfly-cloud-legacy-galleon-pack/wildfly-feature-pack-build.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    <transitive>
```
Comment (jmesnil) [inline wildfly-cloud-legacy-galleon-pack/wildfly-feature-pack-build.xml:25]: related to my question above: should wildfly-datasources-galleon-pack be listed as a dependency?

File: wildfly-modules/jboss/container/wildfly/galleon-wildfly/artifacts/opt/jboss/container/wildfly/galleon/wildfly-s2i-galleon-pack/pom.xml [hunk 6]
```diff
@@ -33,9 +33,10 @@
             <distribution>repo</distribution>
         </license>
     </licenses>
-    
+
     <properties>
         <version.org.wildfly>${env.WILDFLY_VERSION}</version.org.wildfly>
+        <version.org.wildfly.cloud.legacy.galleon-pack>${env.WILDFLY_LEGACY_OS_GALLEON_PACK_VERSION}</version.org.wildfly.cloud.legacy.galleon-pack>
```
Comment (jmesnil) [inline wildfly-modules/jboss/container/wildfly/galleon-wildfly/artifacts/opt/jboss/container/wildfly/galleon/wildfly-s2i-galleon-pack/pom.xml:39]: could you please rename the env to `WILDFLY-CLOUD_LEGACY_GALLEON_PACK_VERSION` to be consistent with the GAV?

File: wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml (hunk 3 above)
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml:3]: There is no explicit dependency on wildfly-datasources-galleon-pack. In order to provision the default configuration a user would provision both wildfly-coud-legacy-galleon-pack and wildfly-datasources-galleon-pack.

The jaxws example provision both feature-packs but that is not related to the default config. Here we are provisioning only layers. You will notice that in the jaxws example we also provision postgresql-driver driver. It is actually not needed, I should remove it.

File: wildfly-cloud-legacy-galleon-pack/wildfly-feature-pack-build.xml (hunk 4 above)
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/wildfly-feature-pack-build.xml:23]: Not a dependency. The 2 FP are separated. Only when provisioning the default config the 2 FPs are required.

File: .gitignore [hunk 7]
```diff
@@ -5,3 +5,4 @@ target
 .idea
 *.iml
 misc
+wildfly-openshift-legacy-galleon-pack/target
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply .gitignore]: Thank-you, fixed

File: wildfly-cloud-legacy-galleon-pack/README.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# wildfly-openshift-legacy-galleon-pack
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/README.md]: Done.

File: wildfly-cloud-legacy-galleon-pack/examples/default-configs/fat-s2i-server/pom.xml [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wildfly.s2i</groupId>
+    <version>1.0.0.Final</version>
+    <artifactId>generate-default-configs</artifactId>
+    <packaging>pom</packaging>
+    <name>Provision aconfig</name>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/examples/default-configs/fat-s2i-server/pom.xml]: Done.

File: wildfly-cloud-legacy-galleon-pack/examples/default-configs/fat-s2i-server/pom.xml [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wildfly.s2i</groupId>
+    <version>1.0.0.Final</version>
+    <artifactId>generate-default-configs</artifactId>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/examples/default-configs/fat-s2i-server/pom.xml]: Thanks fixed.

File: wildfly-cloud-legacy-galleon-pack/examples/default-configs/slim-s2i-server/pom.xml [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wildfly.s2i</groupId>
+    <version>1.0.0.Final</version>
+    <artifactId>generate-default-configs</artifactId>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/examples/default-configs/slim-s2i-server/pom.xml]: Thanks fixed.

File: wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml (hunk 3 above)
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml:3]: No dependency. Both need to provisioned if one wants to provision the default configuration identical to the one present in the image default server.

**Comment by jfdenise** [comment]
@jmesnil , thank-you for your review, I updated the PR with your feedback.

File: wildfly-cloud-legacy-galleon-pack/examples/jaxws/pom.xml [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wildfly.demos</groupId>
+    <artifactId>jaxws-pojo-service</artifactId>
+    <version>1.0.0.Final-SNAPSHOT</version>
+    <packaging>war</packaging>
+    <name>WildFly Bootable JAR - JAX-WS</name>
+    <description>This project is a working example of a JAX-WS web service endpoint; this is the service POM file</description>
```
Comment (yersan) [inline wildfly-cloud-legacy-galleon-pack/examples/jaxws/pom.xml:25]: Just asking, wouldn't be better a JAX-RS example instead? A JAX-RS is easy to test by using the browser / curl

Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/examples/jaxws/pom.xml:25]: The example is just a POC one. It uses currently Bootable JAR Maven plugin, I am using it as a quick way to test the feature-pack manually. The actual examples would be different.

File: wildfly-cloud-legacy-galleon-pack/README.md [hunk 13]
```diff
@@ -0,0 +1,13 @@
+# wildfly-cloud-legacy-galleon-pack
+
+This WildFly Galleon feature-pack captures the changes that are applied to a WildFly server inside the WildFly s2i image.
+
+For now it is used internally by the WildFly s2i image.
+
+# Release the galleon feature-pack
+
+* In the pom.xml file:
+ * Update the dependency on wildFly galleon-pack to the latest WildFly release.
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/README.md]: Thank-you, fixed.

File: wildfly-cloud-legacy-galleon-pack/examples/jaxws/pom.xml [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wildfly.demos</groupId>
+    <artifactId>jaxws-pojo-service</artifactId>
+    <version>1.0.0.Final-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply wildfly-cloud-legacy-galleon-pack/examples/jaxws/pom.xml]: Thank-you, yes.

**Review by jmesnil — APPROVED** [review]

**Pull request merged**